tests/models/test_default_evaluator.py

```diff
@@ -65,15 +68,97 @@ def assert_dict_equal(d1, d2, rtol):
         assert np.isclose(d1[k], d2[k], rtol=rtol)
 
 
-def test_regressor_evaluation(linear_regressor_model_uri, diabetes_dataset):
+def evaluate_model_helper(
+    model,
+    baseline_model,
+    data,
+    targets,
+    model_type: str,
+    dataset_name=None,
+    evaluators=None,
+    evaluator_config=None,
+    eval_baseline_model_only=False,
```

Comment:
In fact, after you implement the disable candidate model eval function, maybe it is possible to pass `model=None` into the `evaluate()` call when the flag ON. In that case you can simplify this function to only take in one model and update the `eval_baseline_model_only` to `eval_as_baseline_model`. Does that make sense?

Reply:
pass model=None will cause the MLflow.evaluate to throw an exception which is one level above the DefaultEvaluator.evaluate